capstone-backend/app.js

```diff
@@ -14,61 +12,148 @@ app.use(cors())
+
   try {
-    const user = await Parse.User.logIn(req.body.username, req.body.password)
-    res.send({"user" : user})
+      await Parse.User.logOut();
+      res.send({ logoutMessage: "User logged out!", typeStatus: "success" });
   } catch (error) {
-    res.status(400)
-    res.send({"error" : "Login failed: " + error })
+      res.send({ logoutMessage: error.message, typeStatus: "danger" });
   }
 })
-// Set up Linkedin OAuth here 
-var router = express.Router();
-const request = require('superagent');
 
-router.get('/profile', function(req, res, next){
-  requestAccessToken(req.body.code, req.query.state)
-  .then(response => {
-    requestProfile(response.body.access_token)
-    .then(response =>{
-      console.log('response: ', response.body);
+app.post('/register', async (req, res) => {
+  Parse.User.enableUnsafeCurrentUser();
+  var currentUser = Parse.User.current();
+      if (currentUser) {
+          Parse.User.logOut();
+      } 
+  const infoUser = req.body;
+  let user = new Parse.User();
+  user.set("username", infoUser.username);
+  user.set("email", infoUser.email);
+  user.set("password", infoUser.password);
+  
 
-      res.render('profile', {profile: response.body})
-    } );
-  })
-  .catch((error) => {
-    res.status(500).send(`${error}`)
-  })
+  try {
+      await user.signUp();
+      await Parse.User.logIn(infoUser.email, infoUser.password);
+      res.send({ signupMessage: "User signed up!", typeStatus: 'success', infoUser: infoUser });
+  }
+  catch (error) {
+      res.send({ signupMessage: error.message, typeStatus: 'danger', infoUser: infoUser });
+  }
 })
+app.post('/profile', async (req, res) => {
+  Parse.User.enableUnsafeCurrentUser();
+  const infoUser = req.body;
 
-function requestAccessToken(code, state){
-  return request.post('https://www.linkedin.com/oauth/v2/accessToken')
-    .send('grant_type=authorization_code')
-    .send('redirect_uri= http://localhost:3000/profile')
-    .send('client_id=86bj0k3tnixpok')
-    .send('client_secret=9XSEs7KoTrAeu78b')
-    .send(`code=${code}`)
-    .send(`state=${state}`)
-}
+  try {
+      const currentUser = Parse.User.current();
+      if (currentUser) {
+          if (infoUser.role && infoUser.role != "") {
+              currentUser.set("role", infoUser.role);
+          }
+          
+          await currentUser.save()
+          res.send({ userInfo: currentUser, saveInfoMessage: "User info saved!", typeStatus: "success", infoUser: infoUser });
+      } else {
+          res.send({ userInfo: "", saveInfoMessage: "No user found", typeStatus: "danger", infoUser: infoUser });
+      }
+  } catch (error) {
+      
+      res.send({ saveInfoMessage: error.message, typeStatus: "danger", infoUser: infoUser });
+  }
+})
+
+app.get('/profile/interests', async(req, res) => {
+  Parse.User.enableUnsafeCurrentUser();
+  const currentUser = Parse.User.current();
+  if (currentUser){
+    const userSkills = await getInterestQuery(currentUser, "Skills");
```

Comment:
Let's keep in mind that we are making a distinction between interests and skills.
So, if this endpoint is named interests, but it's returning skills, then it has the wrong name.